hera_qm/utils.py

```diff
@@ -332,6 +333,45 @@ def generate_fullpol_file_list(files, pol_list):
     # initialize
     file_list = []
 
+    # Check if all input files are full-pol files
+    # if so return the input files as the full list
+    uvd = UVData()
+
+    for filename in files:
+        if filename.split('.')[-1] == 'uvh5':
+            uvd.read_uvh5(filename, read_data=False)
+        else:
+            uvd.read(filename)
+        # convert the polarization array to strings and compare with the
+        # expected input.
+        # If anyone file is not a full-pol file then this will be false.
+        input_pols = uvutils.polnum2str(uvd.polarization_array)
+        full_pol_check = np.array_equal(np.sort(input_pols), np.sort(pol_list))
+
+        if not full_pol_check:
+            # if a file has more than one polarization but not all expected pols
+            # raise an error that mixed pols are not allowed.
+            if len(input_pols) > 1:
+                raise ValueError("The file: {fname} contains {npol} "
+                                 "polarizations: {pol}."
+                                 "Currently only full lists of all expected "
+                                 "polarization files or lists of "
+                                 "files with single polarizations in the "
+                                 "name of the file (e.g. zen.JD.pol.HH.uv) "
+                                 "are allowed.")
```

Comment:
Should there be a `.format()` at the end of this string?

Reply:
Why yes, yes there should.